### DISCUSSION TITLE
whatsapp webhook

### DISCUSSION THREAD

**Discussion by zakkiblue** [spatie/laravel-webhook-client#154] [Q&A]
I tried using it on Whatsapp webhook "The callback URL could not be validated 
![Screenshot (1)](https://user-images.githubusercontent.com/47022280/183325115-edcefdad-0ce8-4471-b28e-70c0a07681a8.png)

and in the ngrok Show error "405 methods not allowed"

![Screenshot (2)](https://user-images.githubusercontent.com/47022280/183325153-02a49f19-1814-4f64-a160-3b1448b577dc.png)

**Comment by praveenkalaiarasu** [comment]
That is because the default macro to create routes using 

```php
Route::webhooks('webhook-receiving-url');
```
creates only a post route. 

While WhatsApp notifications are POST requests, the verification is a GET request

> From WhatsApp Docs:
> Anytime you configure the Webhooks product in your App Dashboard, we'll send a GET request to your endpoint URL. Verification requests include the following query string parameters, appended to the end of your endpoint URL.

EDIT: 

Found a way.
Just add another route of GET type.

```php
Route::
controller(VerificationController::class)
->group(function () {

	Route::get('/webhook-receiving-url', 	'verification')			->name('verification');

});
```

And in controller, verify the `hub_verify_token` value is the same as the one you have set and return a response like, 

```php
namespace App\Http\Controllers;
use Illuminate\Http\Request;
class VerificationController extends Controller
{
	public function verification(Request $request)
	{
		if($request['hub_verify_token'] === 'YOUR_SET_STRING_IN_WHATSAPP_DASHBOARD') {
			return response($request['hub_challenge'], 200);
		} else {
			return response('', 403);
		}
	}
}
```
Mine has been verified:

![image](https://user-images.githubusercontent.com/37588354/190894995-9474ae62-bcca-4744-91a9-b635bd80e17c.png)
